File: 파이썬베이스/부르트포스/투포인터/두수의합.py
import sys
sys.stdin = open('3273.txt', 'r')
input = sys.stdin.readline
N = int(input())
nums = list(map(int, input().split()))
nums.sort()
M = int(input())
result = 0
left = 0
right = N-1

while left < right:
    if nums[left] + nums[right] >= M:
        if nums[left] + nums[right] == M:
            result += 1
        right -= 1
    if nums[left] + nums[right] < M:
        left += 1
print(result)

File: 자스베이스/투포인터/두수의_합.js
// var input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")
var input = require("fs")
  .readFileSync("./3273.txt")
  .toString()
  .trim()
  .split("\r\n");

const n = parseInt(input[0]);
let nums = input[1].split(" ").map(Number);
const x = parseInt(input[2]);
nums = nums.sort((a, b) => a - b);
let left = 0,
  right = n - 1,
  result = 0;
while (left < right) {
  if (nums[left] + nums[right] >= x) {
    if (nums[left] + nums[right] === x) result++;
    right--;
  } else if (nums[left] + nums[right] < x) {
    if (right === n) break;
    left++;
  }
}
console.log(result);
